method.rb

```diff
@@ -0,0 +1,99 @@
+module Enumerable
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_arr = []
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+    new_arr
+  end
+
+  def my_all?
+    stat = true
+    my_each do |num|
+      stat = false unless yield num
+    end
+    stat
+  end
+
+  def my_any?
+    stat = false
+    my_each do |num|
+      stat = true if yield num
+    end
+    stat
+  end
+
+  def my_none?
+    stat = true
+    my_each do |num|
+      stat = false if yield num
+    end
+    stat
+  end
+
+  def my_count(counter = size)
+    p_count = 0
+    if block_given?
+      my_each do |num|
+        p_count += 1 if yield num
+      end
+      counter = p_count
+    end
+    counter
+  end
+
+  def my_inject(acc = 0)
+    my_each do |num|
+      acc = yield(acc, num)
+    end
+    acc
+  end
+
+  def multiply_els
+    my_inject(1) do |acc, num|
+      acc * num
+    end
+  end
+
+  def my_map(prop = nil)
+    new_arr = []
+    if prop
+      my_each do |num|
+        new_arr.push(prop.call(num))
+      end
+
+    else
+      my_each do |num|
+        new_arr.push(yield num)
+      end
+    end
+
+    new_arr
+  end
+end
```

Comment:
- [ ] should have working `my_map` **(covered by automated tests)** identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should return an enumerator if no block or argument is given.